packages/components/nodes/chains/GraphCypherQAChain/GraphCypherQAChain.ts

```diff
@@ -0,0 +1,273 @@
+import {
+    ICommonObject,
+    INode,
+    INodeData,
+    INodeParams,
+    INodeOutputsValue,
+    IServerSideEventStreamer,
+    FlowiseMemory
+} from '../../../src/Interface'
+import { GraphCypherQAChain } from '@langchain/community/chains/graph_qa/cypher'
+import { getBaseClasses } from '../../../src/utils'
+import { BasePromptTemplate, PromptTemplate, FewShotPromptTemplate } from '@langchain/core/prompts'
+import { ConsoleCallbackHandler, CustomChainHandler, additionalCallbacks } from '../../../src/handler'
+import { ConsoleCallbackHandler as LCConsoleCallbackHandler } from '@langchain/core/tracers/console'
+import { checkInputs, Moderation, streamResponse } from '../../moderation/Moderation'
+import { formatResponse } from '../../outputparsers/OutputParserHelpers'
+
+class GraphCypherQA_Chain implements INode {
+    label: string
+    name: string
+    version: number
+    type: string
+    icon: string
+    category: string
+    description: string
+    baseClasses: string[]
+    inputs: INodeParams[]
+    outputs: INodeOutputsValue[]
+    sessionId?: string
+
+    constructor(fields?: { sessionId?: string }) {
+        this.label = 'Graph Cypher QA Chain'
+        this.name = 'graphCypherQAChain'
+        this.version = 2.0
+        this.type = 'GraphCypherQAChain'
+        this.icon = 'graphqa.svg'
+        this.category = 'Chains'
+        this.description = 'Advanced chain for question-answering against a Neo4j graph by generating Cypher statements'
+        this.baseClasses = [this.type, ...getBaseClasses(GraphCypherQAChain)]
+        this.sessionId = fields?.sessionId
+        this.inputs = [
+            {
+                label: 'Language Model',
+                name: 'model',
+                type: 'BaseLanguageModel',
+                optional: true,
+                description: 'Model for generating Cypher queries and answers.'
+            },
+            {
+                label: 'Cypher Generation Model',
+                name: 'cypherModel',
+                optional: true,
+                type: 'BaseLanguageModel',
+                description: 'Model for generating Cypher queries. If not provided, the main model will be used.'
+            },
+            {
+                label: 'QA Model',
+                name: 'qaModel',
+                optional: true,
+                type: 'BaseLanguageModel',
+                description: 'Model for generating answers. If not provided, the main model will be used.'
+            },
+            {
+                label: 'Neo4j Graph',
+                name: 'graph',
+                type: 'Neo4j'
+            },
+            {
+                label: 'Memory',
```

Comment:
I dont think we can have memory here, since the chain itself is not able to accept memory like [conversation chain](https://github.com/FlowiseAI/Flowise/blob/main/packages/components/nodes/chains/ConversationChain/ConversationChain.ts#L262)

Reply:
removed